### ISSUE TITLE
As a user I want a new CSG mesh triangulation algorithm.

### ISSUE THREAD

**Issue by nilsschmidt1337** [nilsschmidt1337/ldparteditor#601]
- [ ] Have to add elemental triangle-triangle intersection testcases (with union, diff, intersection)!

Testcase
```

0 BFC CERTIFY CCW

0 !LPE CSG_QUALITY 16
0 !LPE CSG_CUBOID a1 45 0 0 0 1 0 0 0 1 0 0 0 1
0 !LPE CSG_CUBOID a2 45 2 0 0 1 0 0 0 1 0 0 0 1
0 !LPE CSG_CUBOID a3 45 -2 0 0 1 0 0 0 1 0 0 0 1
0 !LPE CSG_CUBOID b1 2 0 -1 0 .5 0 0 0 .5 0 0 0 .5 
0 !LPE CSG_CUBOID b2 14 2 1 0 .5 0 0 0 .5 0 0 0 .5 
0 !LPE CSG_CUBOID b3 14 -2 1 0 .5 0 0 0 .5 0 0 0 .5 
0 !LPE CSG_DIFFERENCE a1 b1 c1
0 !LPE CSG_UNION a2 b2 c2
0 !LPE CSG_INTERSECTION a3 b3 c3
0 !LPE CSG_COMPILE c1
0 !LPE CSG_COMPILE c2
0 !LPE CSG_COMPILE c3
```

Problematic testcase (split-plane is between other split-planes)

```
0 BFC CERTIFY CCW

0 !LPE CSG_QUALITY 6

0 !LPE CSG_CYLINDER  a  2 0 -1 0 .7 0 0 0 2 0 0 0 .7
0 !LPE CSG_CYLINDER  b  2 -1 0 0 0 2 0 .7 0 0 0 0 .7
0 !LPE CSG_CYLINDER  c  2 0 0 -1 .7 0 0 0 0 .7 0 2 0

0 !LPE CSG_UNION a b d
0 !LPE CSG_UNION d c d

0 !LPE CSG_COMPILE d

```

**Comment by nilsschmidt1337** [comment]
Had to switch to milestone 0.8.38 since 0.8.37 contains too many important bug fixes which can no longer be held back.

**Comment by nilsschmidt1337** [comment]
Closed with commit 53221ad.
